stl/inc/variant

```diff
@@ -636,7 +636,8 @@ struct _Variant_raw_visit1<-1> { // Fallback case for variants too large for any
             return static_cast<_Fn&&>(_Func)(                                                                     \
                 _Variant_tagged_ref_t<_Storage, (n)>{_STD _Variant_raw_get<(n)>(static_cast<_Storage&&>(_Obj))}); \
         }                                                                                                         \
-        _STL_UNREACHABLE
+        _STL_UNREACHABLE;                                                                                         \
+        [[fallthrough]]

```

Comment:
I believe that this will be caught by Clang's `-Wunreachable-code-fallthrough` (not enabled by `-Wall`).

Reply:
I couldn't get `-Wunreachable-code-fallthrough` to repro at all until Sir Walrus on Discord discovered that the warning simply doesn't repro for templates. See https://github.com/llvm/llvm-project/issues/18993#issuecomment-980927985 :

> (we currentyl don't warn on -Wunreachable-code on any templates because there was a concern that performance would be adversely affected by doing CFG work on lots of uninstantiated templates)

(Additionally, Clang doesn't warn about "system headers" by default, so this shouldn't be a concern for official installations of the STL.) If Clang changes its behavior and we encounter this with the GitHub repo in the future, we can suppress the warning, or avoid emitting fallthrough here for Clang.